simulation/executor/simulate_dev.go

```diff
@@ -142,20 +144,65 @@ func (simState *simState) prepareNextSimState(simCtx *simtypes.SimCtx, req abci.
 	simState.pastTimes = append(simState.pastTimes, simState.header.Time)
 	simState.pastVoteInfos = append(simState.pastVoteInfos, req.LastCommitInfo.Votes)
 
+	// increase header height by one
 	simState.header.Height++
 
+	// set header time
 	timeDiff := maxTimePerBlock - minTimePerBlock
 	simState.header.Time = simState.header.Time.Add(
 		time.Duration(minTimePerBlock) * time.Second)
 	simState.header.Time = simState.header.Time.Add(
 		time.Duration(int64(simCtx.GetRand().Intn(int(timeDiff)))) * time.Second)
 
 	// Draw the block proposer from proposers for n+1
-	simState.header.ProposerAddress = simState.nextValidators.randomProposer(simCtx.GetRand())
+	proposerPk := simState.nextValidators.randomProposer(simCtx.GetRand())
+	simState.header.ProposerAddress = proposerPk.Address()
 	// find N + 2 valset
 	nPlus2Validators := updateValidators(simState.tb, simCtx.GetRand(), simState.simParams, simState.nextValidators, res.ValidatorUpdates, simState.eventStats.Tally)
 
 	// now set variables in perspective of block n+1
 	simState.curValidators = simState.nextValidators
 	simState.nextValidators = nPlus2Validators
+
+	var currentValSet tmproto.ValidatorSet
```

Comment:
Thoughts on moving this line and onward into a new function called `simstate.constructHeaderHashes(args)`?

Reply:
Also perhaps we should link to some docs for the TM header?

Reply:
Done on both! https://github.com/osmosis-labs/osmosis/pull/2530/commits/6a2cfed59221483b991093aa8586d2c0296949b4